aemdesign-aem-services/src/main/java/design/aem/utils/components/ComponentDetailsUtil.java

```diff
@@ -459,7 +460,28 @@ public static ComponentProperties getPageInfo(Map<String, Object> pageContext, P
         return componentProperties;
     }
 
-
+    /***
+     * detect of current component being executed by a list component.
+     * @param request request instance to check
+     * @return true if current component being executed by a list
+     */
+    public static boolean isComponentRenderedByList(SlingHttpServletRequest request) {
+        Object includeServlet = request.getAttribute("org.apache.sling.api.include.servlet");
+        if (isNotNull(includeServlet)) {
+            try {
+                Class<?> clazz = includeServlet.getClass();
+                Field field = clazz.getDeclaredField("scriptName"); //Note, this can throw an exception if the field doesn't exist.
+                field.setAccessible(true);
+                String scriptName = (String)field.get(includeServlet);
+                if (scriptName.contains("/components/lists/")) {
+                    return true;
+                }
+            } catch (Exception ex) {
+                return false;
```

Comment:
Do you think is it a better idea to log exception here?

Reply:
nah as there will be a lot of exceptions.